modules/cas_disk/exp_obj.c

```diff
@@ -339,8 +339,32 @@ static void _casdsk_exp_obj_clear_dev_t(struct casdsk_disk *dsk)
 	}
 }
 
+static int _casdsk_exp_obj_open(struct block_device *bdev, fmode_t mode)
+{
+	struct casdsk_disk *dsk = bdev->bd_disk->private_data;
+
+	mutex_lock(&dsk->lock);
```

Comment:
there is a wrapper for that: casdsk_disk_lock()

Reply:
I'm not sure how this interferes with other usages of casdsk_disk_lock(). For example casdsk_disk_set_pt(). I think almost all of the functions that currently call  casdsk_disk_lock() are not used at all. I suggest to remove them and check whether dsk->lock remains used anywhere else at all.